code/modules/antagonists/heretic/status_effects/buffs.dm

```diff
@@ -58,12 +77,23 @@
 
 /datum/status_effect/marshal/on_remove()
 	owner.remove_movespeed_mod_immunities(id, /datum/movespeed_modifier/damage_slowdown)
+	if(!iscarbon(owner))
+		return
+	var/mob/living/carbon/drinker = owner
+	for(var/obj/item/bodypart/potentially_wounded in drinker.bodyparts)
+		for(var/datum/wound/found_wound in potentially_wounded.wounds)
+			found_wound.remove_wound()
```

Comment:
```suggestion
	for(var/datum/wound/found_wound as anything in drinker.all_wounds)
		found_wound.remove_wound()
```

Reply:
I also think this could use a visual or audio effect, kind of like how lings go "crunch" when they regrow stuff. 

Just, instantly healing all wounds and regaining all limbs with no pizzazz feels off.